vaadin-combo-box-flow-parent/vaadin-combo-box-flow/src/main/java/com/vaadin/flow/component/combobox/ComboBoxRenderManager.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2000-2022 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.flow.component.combobox;
+
+import com.vaadin.flow.data.renderer.LitRenderer;
+import com.vaadin.flow.data.renderer.Renderer;
+import com.vaadin.flow.data.renderer.Rendering;
+import com.vaadin.flow.dom.Element;
+import com.vaadin.flow.shared.Registration;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Internal class for managing rendering related logic for combo box components
+ *
+ * @param <TItem>
+ *            Type of individual items that are selectable in the combo box
+ */
+class ComboBoxRenderManager<TItem> implements Serializable {
```

Comment:
I felt that it might be a good thing if we try to separate some concerns, rather than dumping everything into one class. Looking at the diff for `ComboBox` you can kind of see how maintaining a class this way becomes more difficult and leads to related class members being strewn all over the place.
Let me know what you think.